### PR TITLE
FORMS-3624 Data binding for components in Sites editor

### PR DESCRIPTION
**Pull request by dmaurya929**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by rismehta** [comment]
Please add test cases in the PR

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/_cq_dialog/.content.xml [hunk 1]
```diff
@@ -106,7 +106,7 @@
                             </enums>
                             <bindref
                                     jcr:primaryType="nt:unstructured"
-                                    sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
+                                    sling:resourceType="fd/af/authoring/components/granite/form/bindrefselector"
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/_cq_dialog/.content.xml:109]: Is this part of public region, else content classification tool would throw an error

Comment (dmaurya929) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/_cq_dialog/.content.xml:109]: Already using this resource for dorBindRef at line #116 .

**Comment by dmaurya929** [comment]
I have added `schemaRef` and `schemaType` property in example package to tests in sites editor. Please let me know if that is allowed otherwise it will require to create sites collateral in `it` folder.

File: it/config/src/main/content/jcr_root/apps/system/config/com.adobe.granite.toggle.impl.dev.DynamicToggleProviderImpl.cfg.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "enabledToggles":[
-    "ft-cq-4324864"
+    "ft-cq-4324864",
+    "FT_FORMS-2474"
```
Comment (rismehta) [inline it/config/src/main/content/jcr_root/apps/system/config/com.adobe.granite.toggle.impl.dev.DynamicToggleProviderImpl.cfg.json:4]: This is not allowed. It wont' work

**Comment by dmaurya929** [comment]
As discussed @nit23uec , skipping test untill FT_FORMS-2474 will go in pre-release.
Pending: Also insead of adding schemaRef & schemaType in examples, add new page in it folder.
Also please approve changes if looks good

**Review by ujjwal5 — APPROVED** [review]
LGTM
@rismehta can you please review if changes are good to be merged to dev?

**Pull request merged**